### PR TITLE
feat(check:policy): Add policy handler to check for extraneous lockfiles

### PR DESCRIPTION
**Pull request by tylerbutler**
Adds a policy handler that checks that package-lock.json files appear only in the root of release groups or independent packages. Additional paths can be added to the fluidBuild.additionalLockfilePaths array in package.json.

Related to #9956.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -160,6 +160,14 @@
...
-    }
+    },
+    "additionalLockfilePaths": [
+      "tools/telemetry-generator",
+      "common/build/build-common",
+      "common/build/eslint-config-fluid",
+      "docs",
+      "server/gitrest",
+      "server/historian"
+    ]
```
Comment (tylerbutler) [inline package.json:171]:
```suggestion
    "additionalLockfilePaths": [
      "common/build/build-common",
      "common/build/eslint-config-fluid",
      "docs",
      "server/gitrest",
      "server/historian",
      "tools/telemetry-generator"
    ]
```

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts [hunk 2]
```diff
@@ -46,4 +50,44 @@ export const handlers: Handler[] = [
...
+            }
+
+            const returnData: string[] = [];
+            if (path.basename(file) === "package-lock.json") {
+                if (!knownPaths.includes(path.dirname(file))) {
+                    returnData.push(`Unexpected package-lock.json file at: ${file}`);
+                }
+            }
+
+            if (returnData.length > 1) {
```
Comment (curtisman) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts:84]: how can returnData have multiple entries?

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts [hunk 3]
```diff
@@ -46,4 +50,44 @@ export const handlers: Handler[] = [
...
+
+            // Add paths from entries that are arrays
+            const arrayVals = Object.values(manifest.repoPackages)
+                .filter((p) => typeof p !== "string")
+                .map((p) => p);
+            for (const arr of arrayVals) {
+                if (Array.isArray(arr)) {
+                    knownPaths.push(...arr.map((p) => p.toString()));
+                }
+            }
```
Comment (curtisman) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts:75]: The above seems compute the same thing again and again.  Should we cache it?

Comment (tylerbutler) [reply build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts:94]: Refactored this out into a get function that calculates the known paths once and returns the cached value subsequently.

**Review by curtisman — APPROVED** [review]

**Pull request merged**